core/src/mender-client.c

```diff
@@ -1291,20 +1290,14 @@ mender_client_update_work_function(void) {
         continue;                                                                                \
     }
 
-    {
-        cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
-        if (NULL != deployment_id_j) {
-            deployment_id = deployment_id_j->valuestring;
-        }
-    }
     while (MENDER_UPDATE_STATE_END != update_state) {
         switch (update_state) {
             case MENDER_UPDATE_STATE_DOWNLOAD:
                 mender_log_info("Downloading deployment artifact with id '%s', artifact name '%s' and uri '%s'",
                                 deployment->id,
                                 deployment->artifact_name,
                                 deployment->uri);
-                mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_DOWNLOADING);
+                mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_DOWNLOADING);
```

Comment:
This was done on purpose to avoid issues when `deployment` is not initialized above. Keep in mind that this `switch()` can jump into an arbitrary branch.

Reply:
Ah, yes - I didn't think about that